src/main/java/com/uber/kafkaSpraynozzle/KafkaPoster.java

```diff
@@ -179,6 +179,14 @@ private boolean postBatchEvents(List<ByteArrayEntity> batch) {
                 currentUrl = (currentUrl + 1) % urls.size();
             }
             post.setHeader("User-Agent", "KafkaSpraynozzle-0.2.0");
+
+            // we set the timestamp and time cost as if it has failed
+            // So in case it fails we will avoid it in next try;
+            //  Later if we succeeded we will update the value to correct latest value.
+            responseTimestamp[pickedUrlIdx] = System.nanoTime();
```

Comment:
It is just like the "if ... else..." branch, try {} catch{} are more tricky and we prone to miss some of the branches.

If we set a bad value before we do anything, we can be sure that we will only overwrite it with small values when we succeed. All other bad cases we will get a big value.